### ISSUE TITLE
Improv: Similar to AWS AppSync, server should only return fields from user query (filter out extra fields)

### ISSUE THREAD

**Issue by maoosi** [maoosi/prisma-appsync#68]
**Dependency:** Replace dot-object with [wild-wild-path](https://github.com/ehmicky/wild-wild-path/issues/1) in [client/utils](https://github.com/maoosi/prisma-appsync/blob/main/packages/client/src/utils.ts#L1). Wild-wild-path can then be used to filter out query fields based on paths.

**Comment by cipriancaba** [comment]
Was just about to open an issue. Would this fix the situation where I have a custom resolver:

```
type HealthcheckStatus {
  status: String!
  code: Number
}

extend type Query {
  healthcheck: HealthcheckStatus
}
```

Now, when querying the following, the response will include all fields, not filtering just the requested path:
```
query {
  healthcheck {
    status
  }
```

**Comment by maoosi** [comment]
Yes, this is related. Yoga GraphQL returns all the data, regardless of the actual Query. Note that this should only be an issue with the local dev server.

**Comment by maoosi** [comment]
Released in [1.0.0-rc.4](https://github.com/maoosi/prisma-appsync/releases/tag/1.0.0-rc.4).
